code/modules/projectiles/ammo_datums.dm

```diff
@@ -2202,16 +2202,27 @@ datum/ammo/bullet/revolver/tp44
 		mob_caught.IgniteMob()
 
 /datum/ammo/energy/plasma_pistol/on_hit_mob(mob/M, obj/projectile/proj)
-	var/turf/T = get_turf(M)
-	if(!T)
-		T = get_turf(proj)
-	T.ignite(heat, burn_damage, fire_color)
+	var/turf/target = get_turf(M)
+	if(!target)
+		target = get_turf(proj)
+	target.ignite(heat, burn_damage, fire_color)
 
 /datum/ammo/energy/plasma_pistol/on_hit_obj(obj/O, obj/projectile/proj)
-	var/turf/T = get_turf(O)
-	if(!T)
-		T = get_turf(proj)
-	T.ignite(heat, burn_damage, fire_color)
+	var/turf/target = get_turf(O)
+	if(!target)
+		target = get_turf(proj)
+	target.ignite(heat, burn_damage, fire_color)
+
+/datum/ammo/energy/plasma_pistol/do_at_max_range(turf/T, obj/projectile/proj)
+	var/burn_mod = 1
+	if(istype(T, /turf/closed/wall))
+		burn_mod = 3
```

Comment:
I don't like this duplicating code from pp's turf hit. Better off using a drop_flame